src/renderer/logger.ts

```diff
@@ -0,0 +1,56 @@
+import { log } from '../common/utils';
+
+// API Calls Logger
+
+const handler = {
```

Comment:
How is this logger different than a classic logger or simply debug?

Reply:
Ah it's not a classic logger! It's a spy!

Reply:
This logger proxy access to the given context and log each call event for APIs not implemented with the result and params. The `console.debug` needs to be added in each function declaration. This middleware is better at the usage for debug functions call and event dispatch than inline debug call

Reply:
I think a better name would be `SpyLoger` or `LoggingProxy`

Reply:
Renamed for `LoggingProxy `